storage/ctab/src/main/java/org/openscience/cdk/io/CharIter.java

```diff
@@ -0,0 +1,100 @@
+package org.openscience.cdk.io;
+
+/**
+ * Helper class that facilitates the parsing of text data.
+ */
+final class CharIter {
```

Comment:
Is this not available from default Java?

Reply:
I don't think the functionality provided by this class is available from Java (e.g, look-ahead).

Reply:
It is not, basically it provides some look ahead/consume for parsing manually.